gguf-py/gguf/gguf_writer.py

```diff
@@ -608,6 +735,46 @@ def _pack_val(self, val: Any, vtype: GGUFValueType, add_vtype: bool) -> bytes:
 
         return kv_data
 
-    def _write_packed(self, fmt: str, value: Any, skip_pack_prefix: bool = False) -> None:
-        assert self.fout is not None
-        self.fout.write(self._pack(fmt, value, skip_pack_prefix))
+    def _write_packed(self, fout: BufferedWriter, fmt: str, value: Any, skip_pack_prefix: bool = False) -> None:
+        assert fout is not None
+        fout.write(self._pack(fmt, value, skip_pack_prefix))
+
+    @staticmethod
+    def get_tensor_size(tensor) -> int:
+        try:
+            return tensor.data_type.elements_to_bytes(np.prod(tensor.shape))
+        except AttributeError: # numpy ndarray[Any, Any]
+            return tensor.nbytes
```

Comment:
With `TensorInfo`, the size in bytes is always available with the `nbytes` field, so this function doesn't seem necessary.